src/v/model/record.h

```diff
@@ -350,6 +350,17 @@ class record_batch_attributes final {
     std::bitset<16> _attributes;
 };
 
+struct key_offset {
```

Comment:
maybe we can change this name to in the `record_...` convention, we have `record`, `record_batch`, maybe `record_key_offset` ?